### PR TITLE
Add initial designs for campaign table

### PR DESCRIPTION
**Pull request by jjolmo**
Initial approach of the campaigns table redesign in BlazePress

image

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/promote-post-i2/components/campaign-item/index.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+			<td>
+				<Badge type={ getCampaignStatusBadgeColor( status ) }>
+					{ getCampaignStatus( status ) }
+				</Badge>
+			</td>
+			<td>{ getCampaignEndText( moment, campaign.status, campaign.end_date ) }</td>
+			<td>{ budgetString }</td>
+			<td>{ formatNumber( impressions_total ) }</td>
+			<td>{ formatNumber( clicks_total ) }</td>
+			<td>{ campaignContainsData && <Button icon={ chevronRight } onClick={ () => {} } /> }</td>
```
Comment (timur987) [inline client/my-sites/promote-post-i2/components/campaign-item/index.tsx]: TeamCity said "Unexpected empty arrow function. (@typescript-eslint/no-empty-function)"

https://teamcity.a8c.com/viewLog.html?buildId=9841488&tab=Inspection&buildTypeId=calypso_calypso_WebApp_Check_Code_Style_Branch

File: client/my-sites/promote-post-i2/components/campaigns-list/index.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+				<SitePlaceholder />
+			</div>
+		);
+	}
+
+	if ( ! memoCampaigns.length ) {
+		return <EmptyPromotionList type="campaigns" />;
+	}
+
+	const handleClickCampaign = ( isExpanded: boolean, campaign: Campaign ) => {
```
Comment (timur987) [inline client/my-sites/promote-post-i2/components/campaigns-list/index.tsx]: Probably it's work in progress, IDE says that the `handleClickCampaign` variable is unused.

**Comment by jjolmo** [comment]
Yeah. I'm aware this is unfinished. Maybe I can upload a better status of this PR. But I think it's not that bad to iterate on this WIP

**Comment by jjolmo** [comment]
Done! I've fixed these warnings so we don't block merging this

**Review by timur987 — APPROVED** [review]
Thanks, @jjolmo!

Overall changes looks good and works as expected.
By the way, I found the way how to improve it a bit. Feel free to use/ignore this [patch.txt](https://github.com/Automattic/wp-calypso/files/11252969/patch.txt).

The result of patch's application can looks like this:
campaign-redesign-suggestion

What it does is:
- adjust heights of the search bar and switches to the right,
- remove white background after switchers,
- adjust font sizes,
- removes word wrapping for campaign badge.

Approving by assumption you will apply the patch of its analog to fix the current state that looks like this:
image

**Comment by jjolmo** [comment]
I've applied the patch. Thanks @timur987 !

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7932493

Thank you @jjolmo for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7932493).

**Pull request merged**